pkg/kwokctl/cmd/kwokctl/create/cluster/cluster.go

```diff
@@ -0,0 +1,202 @@
+	cmd.Flags().Uint32Var(&flags.PrometheusPort, "prometheus-port", uint32(vars.PrometheusPort), `port to expose Prometheus metrics`)
+	cmd.Flags().BoolVar(&flags.SecurePort, "secure-port", vars.SecurePort, `apiserver use TLS`)
+	cmd.Flags().BoolVar(&flags.QuietPull, "quiet-pull", vars.QuietPull, `pull without printing progress information`)
+	cmd.Flags().StringVar(&flags.EtcdImage, "etcd-image", vars.EtcdImage, `image of etcd, only for docker/nerdctl runtime
+'${KWOK_KUBE_IMAGE_PREFIX}/etcd:${KWOK_ETCD_VERSION}'
+`)
+	cmd.Flags().StringVar(&flags.KubeApiserverImage, "kube-apiserver-image", vars.KubeApiserverImage, `image of kube-apiserver, only for docker/nerdctl runtime
+'${KWOK_KUBE_IMAGE_PREFIX}/kube-apiserver:${KWOK_KUBE_VERSION}'
+`)
+	cmd.Flags().StringVar(&flags.KubeControllerManagerImage, "kube-controller-manager-image", vars.KubeControllerManagerImage, `image of kube-controller-manager, only for docker/nerdctl runtime
+'${KWOK_KUBE_IMAGE_PREFIX}/kube-controller-manager:${KWOK_KUBE_VERSION}'
+`)
+	cmd.Flags().StringVar(&flags.KubeSchedulerImage, "kube-scheduler-image", vars.KubeSchedulerImage, `image of kube-scheduler, only for docker/nerdctl runtime
+'${KWOK_KUBE_IMAGE_PREFIX}/kube-scheduler:${KWOK_KUBE_VERSION}'
+`)
+	cmd.Flags().StringVar(&flags.KwokControllerImage, "kwok-controller-image", vars.KwokControllerImage, `image of kwok, only for docker/nerdctl/kind runtime
+'${KWOK_IMAGE_PREFIX}/kwok:${KWOK_VERSION}'
+`)
+	cmd.Flags().StringVar(&flags.PrometheusImage, "prometheus-image", vars.PrometheusImage, `image of Prometheus, only for docker/nerdctl/kind runtime
+'${KWOK_PROMETHEUS_IMAGE_PREFIX}/prometheus:${KWOK_PROMETHEUS_VERSION}'
+`)
+	cmd.Flags().StringVar(&flags.KindNodeImage, "kind-node-image", vars.KindNodeImage, `image of kind node, only for kind runtime
+'${KWOK_KIND_NODE_IMAGE_PREFIX}/node:${KWOK_KUBE_VERSION}'
+`)
+	cmd.Flags().StringVar(&flags.KubeApiserverBinary, "kube-apiserver-binary", vars.KubeApiserverBinary, `binary of kube-apiserver, only for binary runtime
+'${KWOK_KUBE_BINARY_PREFIX}/kube-apiserver'
+`)
+	cmd.Flags().StringVar(&flags.KubeControllerManagerBinary, "kube-controller-manager-binary", vars.KubeControllerManagerBinary, `binary of kube-controller-manager, only for binary runtime
+'${KWOK_KUBE_BINARY_PREFIX}/kube-controller-manager'
+`)
+	cmd.Flags().StringVar(&flags.KubeSchedulerBinary, "kube-scheduler-binary", vars.KubeSchedulerBinary, `binary of kube-scheduler, only for binary runtime
+'${KWOK_KUBE_BINARY_PREFIX}/kube-scheduler'
+`)
+	cmd.Flags().StringVar(&flags.KwokControllerBinary, "kwok-controller-binary", vars.KwokControllerBinary, `binary of kwok-controller, only for binary runtime
+`)
+	cmd.Flags().StringVar(&flags.EtcdBinaryTar, "etcd-binary-tar", vars.EtcdBinaryTar, `tar of etcd, only for binary runtime
+`)
+	cmd.Flags().StringVar(&flags.PrometheusBinaryTar, "prometheus-binary-tar", vars.PrometheusBinaryTar, `tar of Prometheus, only for binary runtime
```

Comment:
Is `tar` intentional? I think we should accept binary path as the input, either to replace `tar` or coexist.

And an 💭 is to lookup the default path of etcd/prometheus and use it by default, if not specified.

Reply:
for Mac M1, there is no prebuild etcd image/binary lol...

https://github.com/etcd-io/etcd/releases/tag/v3.5.4

- [etcd-v3.5.4-darwin-amd64.zip](https://github.com/etcd-io/etcd/releases/download/v3.5.4/etcd-v3.5.4-darwin-amd64.zip)
- [etcd-v3.5.4-linux-amd64.tar.gz](https://github.com/etcd-io/etcd/releases/download/v3.5.4/etcd-v3.5.4-linux-amd64.tar.gz)
- [etcd-v3.5.4-linux-arm64.tar.gz](https://github.com/etcd-io/etcd/releases/download/v3.5.4/etcd-v3.5.4-linux-arm64.tar.gz)
- [etcd-v3.5.4-linux-ppc64le.tar.gz](https://github.com/etcd-io/etcd/releases/download/v3.5.4/etcd-v3.5.4-linux-ppc64le.tar.gz)
- [etcd-v3.5.4-linux-s390x.tar.gz](https://github.com/etcd-io/etcd/releases/download/v3.5.4/etcd-v3.5.4-linux-s390x.tar.gz)
- [etcd-v3.5.4-windows-amd64.zip](https://github.com/etcd-io/etcd/releases/download/v3.5.4/etcd-v3.5.4-windows-amd64.zip)

Reply:
Is tar intentional? I think we should accept binary path as the input, either to replace tar or coexist.
Etcd and Prometheus releases do not provide a binary

        And an 💭 is to lookup the default path of etcd/prometheus and use it by default, if not specified.

That makes sense, I'll see how to do it right.

Reply:
We can do it in a follow-up.

Pretty nice, I can install it on my M1 🎉 :

```
⇒  kwokctl create cluster --runtime binary --kube-scheduler-binary /usr/local/bin/kube-scheduler@1.24.3 --kube-controller-manager-binary /usr/local/bin/kube-controller-manager@1.24.3  --kube-apiserver-binary /usr/local/bin/kube-apiserver@1.24.3 --kwok-controller-binary /usr/local/bin/kwok --etcd-binary /opt/homebrew/bin/etcd
Creating cluster "kwok-kwok"
Starting cluster "kwok-kwok"
Cluster "kwok-kwok" is ready
You can now use your cluster with:

    kubectl config use-context kwok-kwok

Thanks for using kwok!
```